ci/jenkins/stop-jobs.sh

```diff
@@ -0,0 +1,156 @@
+
+    for (( j=0; j<$length_of_queue; j++ )); do
+        job_name=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].task.name" | tr -d '"') 
+        if [[ "$job_name" == "$capv_job_name" ]]; then
+            # fetch the waiting job id and its pr id.
+            job_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].id")
+            waiting_job_pr_id=$(curl -s -XGET "${JENKINS_URL}queue/api/json" | jq "[.[]][2][$j].actions[1].parameters[9].value" | tr -d '"')
+
+            if [[ "$PULL_REQUEST_ID" == "$waiting_job_pr_id" ]]; then
+                echo "====== abort waiting job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+                curl -XPOST "${JENKINS_URL}/queue/cancelItem?id=$job_id" -u admin:$JENKINS_API_TOKEN
+                break
+            fi
+        fi
+    done
+}
+
+function abort_running_jobs {
+    echo "====== checking for capv stale running jobs on PR ${PULL_REQUEST_ID} ======"
+    capv_job_name=$1
+    i=$2
+    # job color is the status of the jobs, for example if jobs are in running state then its value will be {"red_anime","blue_anime","aborted_anime".....}
+    # so here "*_anime" means job is in running state.
+    # red_anime: last failed job is in running state.
+    # blue_anime: last successful job is in running state.
+    # aborted_anime: last aborted job is in running state.   
+    job_color=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].color" | tr -d '"')
+
+    if [[ ${job_color: -5} == "anime" ]]; then
+        last_build_number=$(curl -s -XGET "${JENKINS_URL}job/$capv_job_name/api/json" | jq "[.[]][10][0].number" | tr -d '"')
+        pr_id=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].ghprbPullId"| tr -d '"')
+
+        if [[ "$PULL_REQUEST_ID" == "$pr_id" ]]; then
+            echo "====== abort running job $capv_job_name on PR ${PULL_REQUEST_ID} ======"
+
+            # need to stop first then clean.
+            curl -XPOST "${JENKINS_URL}job/$capv_job_name/$last_build_number/stop" -u admin:$JENKINS_API_TOKEN
+
+            CLUSTER_NAME=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].BUILD_TAG"| tr -d '"')
+            ip_of_capv_testbed=$(curl -s -XGET "${JENKINS_URL}job/"$capv_job_name"/"$last_build_number"/injectedEnvVars/api/json" | jq "[.[]][1].SSH_CONNECTION" | awk '{print $3}')
+            
+            cleanup_running_job "$CLUSTER_NAME" "$ip_of_capv_testbed"
+        fi
+    fi   
+}
+
+function stop_kill() {
+    echo "====== checking for stale jobs on PR ${PULL_REQUEST_ID} ======"
+    length_of_capv_jobs=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10]" | jq length)
+
+    for ((i=0; i<$length_of_capv_jobs; i++)); do
+        capv_job_name=$(curl -s -XGET "${JENKINS_URL}label/antrea-test-node/api/json" | jq "[.[]][10][$i].name" | tr -d '"')
```

Comment:
Better to save the curl command output in a variable instead of curl the same URL each time in the loop. This avoid unnecessary API calls to Jenkins.